plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1798,7 +1799,7 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
          if( !remove_expired_blacklisted_trxs( preprocess_deadline ) )
             return start_block_result::exhausted;
          if( !_subjective_billing.remove_expired( _log, chain.pending_block_time(), fc::time_point::now(),
-                                                  [&](){ return should_interrupt_start_block( preprocess_deadline ); } ) ) {
+                                                  [&](){ return should_interrupt_start_block( preprocess_deadline, pending_block_num ); } ) ) {
```

Comment:
Here we have assigned `pending_block_num = hbs->block_num + 1;`.

Is it the same as `chain.pending_block_num();`?

Reply:
Yes. Head is what we are building on and pending is what we are building. Therefore, pending is always head+1.

Reply:
> Head is what we are building on and pending is what we are building

Hum, that would mean head = pending. What is head really? The last block we know about?

Reply:
Let me try that again. `[block#3][block#4 head][pending...` block#4 is a complete block. It is the head of the chain. pending is an incomplete block we are currently building.

Reply:
Oh, thanks, got you, I missed the `on` in `building on` in my first reading.